rfcs/0168-feature-parameter-names.md

```diff
@@ -0,0 +1,559 @@
+---
+feature: feature_parameter_names
+start-date: 2024-01-10
+author: Dmitry Bogatov
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Nixpkgs usually exports compile-time options of the upstream build system
+through the named parameters of derivation function, but does it inconsistent
+way.
+
+# Motivation
+[motivation]: #motivation
+
+Developer interface for building environment with specific features enabled and
+disabled is more complicated and requires more knowledge of implementation
+details compared to already existing systems.
+
+For example, Gentoo Linux has exhaustive list of [all known configuration
+flags](https://www.gentoo.org/support/use-flags) and has means to
+programmatically query what flags are available for particular package. Nixpkgs
+has neither, and this RFC strives to rectify it.
+
+# Detailed design
+[design]: #detailed-design
+
+When upstream build system has options to enable or disable a optional features
+or configure some compile-time parameter, like path to default mailbox or
+initial size of hash map, derivation function (for short, package) MAY expose
+compile-time options via feature parameters that match
+`^(with|conf|enable)_[a-z_0-9]$` regular expression. For short, we will refer
+to them as `with`-parameter, `conf`-parameter and `enable`-parameter correspondingly,
+collectively as `feature parameters`.
+
+Note that this RFC does not establish any requires whether package should or
+should not have any feature parameters, this decision is left at descretion of
+the package maintainer(s). This RFC only concerns itself with the naming of the
+feature flags. Further text will use wording "corresponding feature parameter"
+with implicit caveat "provided that package maintainer decided to export
+underlying build system configuration option".
+
+Feature parameters MUST only be used according to the rules outlined in this
+RFC. They MUST NOT be used for any other purpose.
+
+Boolean feature parameter MUST be either `with`-parameter or `enable`-parameter.
+Non-boolean (e.g. string or integer) feature parameter MUST be an `conf`-parameter.
+
+Feature paramter that incurs additional runtime dependency MUST be and `with`-parameter.
```

Comment:
This line needs rewriting.

Reply:
What exactly is wrong with it?

Reply:
It looks like «and» is a remainder of some previous sentence structure? Of course «parmter» is a typo, but I hope it was «parameter».

Reply:
`s/and/a/`